File: ios/Classes/Event/AdEvent.m
//
//  AdEvent.m
//  flutter_qq_ads
//
//  Created by zero on 2021/8/13.
//

#import "AdEvent.h"

@implementation AdEvent

- (id)initWithAdId:(NSString *)adId andAction:(NSString *)action{
    self.adId=adId;
    self.action=action;
    return self;
}

- (NSDictionary *)toMap{
    NSDictionary *data=@{@"adId":_adId,@"action":_action};
    return data;
}

@end

File: android/src/main/java/com/zero/flutter_pangle_ads/event/AdEvent.java
package com.zero.flutter_pangle_ads.event;

import java.util.HashMap;

/**
 * 广告事件
 */
public class AdEvent {
    // 广告 id
    private final String adId;
    // 操作
    private final String action;

    public AdEvent(String adId, String action) {
        this.adId = adId;
        this.action = action;
    }

    /**
     * 转换为 Map 方面传输
     * @return 转换后的 Map 对象
     */
    public HashMap<String,Object> toMap() {
        HashMap<String,Object> map=new HashMap<String,Object>();
        map.put("adId",adId);
        map.put("action",action);
        return map;
    }
}
